react-components/src/components/CacheProvider/requests.ts

```diff
@@ -54,28 +55,59 @@ export async function inspectNodes(
 }
 
 export async function treeIndexesToNodeIds(
-  modelId: number,
-  revisionId: number,
-  treeIndexes: number[],
+  modelId: ModelId,
+  revisionId: RevisionId,
+  treeIndexes: TreeIndex[],
   cogniteClient: CogniteClient
-): Promise<number[]> {
+): Promise<NodeId[]> {
   const outputsUrl = `${cogniteClient.getBaseUrl()}/api/v1/projects/${
     cogniteClient.project
   }/3d/models/${modelId}/revisions/${revisionId}/nodes/internalids/bytreeindices`;
-  const response = await cogniteClient.post<{ items: number[] }>(outputsUrl, {
-    data: { items: treeIndexes }
+
+  const treeIndexChunks = chunk(treeIndexes, 1000);
+
+  const nodeIds: NodeId[] = [];
+
+  for (const treeIndexChunk of treeIndexChunks) {
+    const response = await cogniteClient.post<{ items: NodeId[] }>(outputsUrl, {
+      data: { items: treeIndexChunk }
+    });
+
+    if (response.status === 200) {
+      nodeIds.push(...response.data.items);
+    } else {
+      throw Error(
+        `treeIndex-nodeId translation failed for treeIndexes ${treeIndexChunk.join(',')}`
+      );
+    }
+  }
+
+  return nodeIds;
+}
+
+export async function nodeIdsToTreeIndices(
+  modelId: ModelId,
+  revisionId: RevisionId,
+  nodeIds: NodeId[],
+  cogniteClient: CogniteClient
+): Promise<NodeId[]> {
```

Comment:
As we are returning treeindices, can we use the TreeIndex type instead?

Reply:
Definitely, good catch :+1: